x/auth/ante/validator_tx_fee.go

```diff
@@ -41,18 +41,19 @@ func checkTxFeeWithValidatorMinGasPrices(ctx sdk.Context, tx sdk.Tx) (sdk.Coins,
 		}
 	}
 
-	priority := getTxPriority(feeCoins)
+	priority := getTxPriority(feeCoins, int64(gas))
 	return feeCoins, priority, nil
 }
 
 // getTxPriority returns a naive tx priority based on the amount of the smallest denomination of the fee
 // provided in a transaction.
-func getTxPriority(fee sdk.Coins) int64 {
+func getTxPriority(fee sdk.Coins, gas int64) int64 {
 	var priority int64
 	for _, c := range fee {
 		p := int64(math.MaxInt64)
-		if c.Amount.IsInt64() {
-			p = c.Amount.Int64()
+		price := c.Amount.QuoRaw(gas)
+		if price.IsInt64() {
+			p = price.Int64()
 		}
 		if priority == 0 || p < priority {
```

Comment:
would be worth to add a NOTE that this is really a demo and should be used with clear understanding of the ordering.

Reply:
I'm not sure what to write, do you mean recommending chains to customize this logic?

Reply:
I'm not following your comment @robert-zaremba. The method's godoc should describe everything we need.

Reply:
I think the naive implementation is not safe when combined with other coins (eg for tx Tips). In general _naive_ doesn't mean bad. But here _naive_ should be really used with deeper consideration. So my suggestion was to make it more clear. 

How about:
_NOTE: This implementation should be used with a great consideration as it opens potential attack vectors where txs with multiple coins could not be prioritize as expected_

Reply:
That's fair. A simple and clear godoc should suffice. @yihuang can you make a quick tweak to the method's godoc pls?

Reply:
sounds good, done.